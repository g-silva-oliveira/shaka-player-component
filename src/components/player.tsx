import React, { useRef } from 'react';
import * as Hooks from '../hooks';

import { IPlayerProps } from '../types';

const ReactPlayer = (props: IPlayerProps) => {
  const videoRef = useRef<HTMLVideoElement | null>(null);
  const uiContainerRef = useRef<HTMLDivElement | null>(null);

  const { player, ui } = Hooks.usePlayer(videoRef, uiContainerRef, props);
  Hooks.usePlayerListener(player, props);
  Hooks.useUIListener(ui, player, props);
  Hooks.useStats(player, props);

  const {
    className,
    config,
    onBuffering,
    onEnded,
    onLoad,
    onPause,
    onPlay,
    onPlayerError,
    onStatsChange,
    playerClassName,
    superConfig,
    uiConfig,
    ...newProps
  } = props;

  const style = {
    maxWidth: "100%",
    width: "100%",
  };

  const overlayClassName = className === undefined ? "mk-theme" : "mk-theme " + props.className;

  return (
    <div style={style} ref={uiContainerRef} className={overlayClassName}>
      <video
        ref={videoRef}
        className={props.playerClassName}
        style={style}
        {...newProps}
      />
    </div>
  );
};

export { ReactPlayer };
