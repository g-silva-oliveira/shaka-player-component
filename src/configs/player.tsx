const vodConfig = {
  ui: {
    controlPanelElements: ["play_pause", "time_and_duration", "mute", "fullscreen"]
  },
};

const streamingConfig = {
  player: {
    manifes: {
      dash: {
        ignoreMinBufferTime: true,
      },
    },
    streaming: {
      lowLatencyMode: true,
      inaccurateManifestTolerance: 0,
      rebufferingGoal: 1,
      smallGapLimit: 1,
      jumpLargeGaps: true,
      durationBackoff: 0
    },
  },
  ui: {
    addSeekBar: false,
    controlPanelElements: ["time_and_duration", "mute", "fullscreen"],
  },
};

export { vodConfig, streamingConfig };
