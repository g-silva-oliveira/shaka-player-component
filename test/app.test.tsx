import * as Enzyme from 'enzyme';
import * as React from 'react';
import * as renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import { ReactPlayer as Player } from '../src/components/player';

Enzyme.configure({ adapter: new Adapter() });

describe("Player", () => {
  it("Should render correctly", () => {
    const tree = renderer.create(<Player />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Should renders source[manifest] string correctly", () => {
    const props = { src: "this is sample src" };
    const tree = Enzyme.shallow(<Player {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it("Should render className container correctly", () => {
    const props = { className: "container-class-name" };
    const tree = Enzyme.shallow(<Player {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it("Should render playerClassName container correctly", () => {
    const props = { playerClassName: "player-class-name" };
    const tree = Enzyme.shallow(<Player {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it("Should render container className and playerClassName container correctly", () => {
    const props = {
      className: "container-class-name",
      playerClassName: "player-class-name",
    };
    const tree = Enzyme.shallow(<Player {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
