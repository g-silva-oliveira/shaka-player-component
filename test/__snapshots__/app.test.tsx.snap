// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player Should render className container correctly 1`] = `ShallowWrapper {}`;

exports[`Player Should render container className and playerClassName container correctly 1`] = `ShallowWrapper {}`;

exports[`Player Should render correctly 1`] = `
<div
  className="mk-theme"
  style={
    {
      "maxWidth": "100%",
      "width": "100%",
    }
  }
>
  <video
    style={
      {
        "maxWidth": "100%",
        "width": "100%",
      }
    }
  />
</div>
`;

exports[`Player Should render playerClassName container correctly 1`] = `ShallowWrapper {}`;

exports[`Player Should renders source[manifest] string correctly 1`] = `ShallowWrapper {}`;
